src/Domain/Exception/UnknownCollectionElementException.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\Domain\Exception;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+final class UnknownCollectionElementException extends \RuntimeException implements DomainExceptionInterface
```

Comment:
It should extends [`\OutOfBoundsException`](https://secure.php.net/manual/fr/class.outofboundsexception.php) instead.

```suggestion
final class UnknownCollectionElementException extends \OutOfBoundsException implements DomainExceptionInterface
```